src/main/java/Spike.java

```diff
@@ -0,0 +1,217 @@
+import Exceptions.TaskNotFoundException;
+import Tasks.Deadline;
+import Tasks.Event;
+import Tasks.Task;
+import Tasks.Todo;
+import Exceptions.ArgumentNotFoundException;
+import Exceptions.CommandNotFoundException;
+
+
+import java.util.ArrayList;
+import java.io.FileNotFoundException;
+import java.io.IOException;
+import java.util.Scanner;
+
+public class Spike {
+    private static final String CHATBOT = "Spike";
+    public static final String DIVIDER = "_________________________________________________";
+    public static final int MARK_TASK_INDEX = 5;
+    public static final int UNMARK_TASK_INDEX = 7;
+    public static final int EVENT_TASK_INDEX = 6;
+    public static final int DEADLINE_TASK_INDEX = 9;
+    public static final int TODO_TASK_INDEX = 5;
+    public static final int DELETE_TASK_INDEX = 7;
+
+    public static void main(String[] args) throws IOException {
+        displayWelcomeMsg();
+
+
+        ArrayList<Task> inputList = DataHandler.readFileContents(DataHandler.FILE_PATH);
+        Scanner in = new Scanner(System.in);
+
+        while (true) {
+            try {
+                startChatbot(in, inputList);
+                displayByeMsg();
+                break;
+            } catch (CommandNotFoundException e) {
+                System.out.println("Command Not Found! Please Try Again");
+            } catch (ArgumentNotFoundException e) {
+                System.out.println("Argument Not Found! Please Try Again");
+            } catch (TaskNotFoundException e){
+                System.out.println("Task Not Found! Please Try Again");
+            } catch (IOException e) {
+                throw new RuntimeException(e);
+            }
+        }
+    }
+
+
+    private static void startChatbot(Scanner in, ArrayList<Task> inputList)
+            throws CommandNotFoundException, ArgumentNotFoundException, TaskNotFoundException, IOException {
+        outerLoop:
+        while (true) {
+            String input = in.nextLine();
+            switch (input.split(" ")[0]) {
```

Comment:
Can consider abstracting this to a parser class in the future similar to personsbook